camp/camp-brooklyn/src/main/java/org/apache/brooklyn/camp/brooklyn/spi/dsl/methods/DslComponent.java

```diff
@@ -353,6 +386,33 @@ public DslSensorSupplier(DslComponent component, Object sensorIndicator) {
         }
 
         @Override
+        public Maybe<Sensor<?>> getImmediately() {
+            return getImmediately(sensorName, false);
+        }
+        
+        protected Maybe<Sensor<?>> getImmediately(Object si, boolean resolved) {
+            if (si instanceof Sensor) {
+                return Maybe.<Sensor<?>>of((Sensor<?>)si);
+            } else if (si instanceof String) {
+                Entity targetEntity = component.get();
+                Sensor<?> result = null;
+                if (targetEntity!=null) {
+                    result = targetEntity.getEntityType().getSensor((String)si);
+                }
+                if (result!=null) return Maybe.<Sensor<?>>of(result);
+                return Maybe.<Sensor<?>>of(Sensors.newSensor(Object.class, (String)si));
```

Comment:
Not sure I follow. In `attributeWhenReady`, if we're given a sensor name then we will instantiate a new sensor object if it's not defined on the entity type; and then we look up the sensor's value (returning Maybe.absent if it's null). Here, we just want it to return the sensor object (rather than a sensor value) so we do the first half: instantiate a new sensor object if it's not defined on the entity type.

This is consistent with the blocking `get()` method. We return `Maybe.absent()` if the `get()` method would have blocked, and otherwise we return the same value as `get()` would have returned.